src/main/java/algorithms/curated170/hard/WordSquaresTrieBacktracking.java

```diff
@@ -0,0 +1,93 @@
+package algorithms.curated170.hard;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class WordSquaresTrieBacktracking {
+   public static class Solution {
+       public final static int LETTER_a = 97;
+
+       public List<List<String>> wordSquares(String[] words) {
+           TrieNode root = buiildTrie(words);
+           int n = words[0].length();
+           TrieNode[] trieNodeState = new TrieNode[n];
+           Arrays.fill(trieNodeState, root);
+           List<List<String>> ans = new ArrayList<>();
+           dfs(0, new char[n][n], ans, trieNodeState);
+           return ans;
+       }
+       void dfs(int rowIndex, char[][] matrix, List<List<String>> ans, TrieNode[] trieNodeState) {
+           if (rowIndex == matrix.length) {
+               ans.add(convertMatrixToList(matrix, rowIndex));
+               return;
+           }
+           TrieNode[] trieNodeStateNext = new TrieNode[trieNodeState.length];
+           for (String candidate : trieNodeState[rowIndex].list) {
+
+               if (isSymmetric(matrix, candidate, rowIndex, trieNodeState, trieNodeStateNext)){
```

Comment:
Note: `isSymmetric()` method handles the backtracking part too. As we call the method, it overrides the character matrix with candidate string. If it's okay, we continue with it. If it's not symmetric, we don't. Either way, it get's overriden in the next function call.